### PR TITLE
Provide default gateway setup

### PR DESCRIPTION
**Pull request by solmonk**
**What type of PR is this?**

feature

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

* Add DEFAULT_SERVICE_NETWORK env variable, on cluster startup create service network and associate VPC
* Keeps CLUSTER_LOCAL_GATEWAY behavior (for conformance test) with new network override mode, unlike CLUSTER_LOCAL_GATEWAY, register all gateways to the same default service network.

**If an issue # is not available please add repro steps and logs from aws-gateway-controller showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this PR introduce any new dependencies?**:

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/configure/environment.md [hunk 1]
```diff
@@ -55,84 +55,23 @@ When set as "debug", the AWS Gateway API Controller will emit debug level logs.
...
 
-When it is set to something different from "NO_DEFAULT_SERVICE_NETWORK", the AWS Gateway API Controller will associate its correspoding Lattice Service Network to cluster's VPC.
-Also, all HTTPRoutes will automatically have an additional parentref to this `CLUSTER_LOCAL_GATEWAY`.
+When set as a non-empty value, creates a service network with that name.
+The created service network will be also associated with cluster VPC.
 
 ---
 
-#### `TARGET_GROUP_NAME_LEN_MODE`
+#### `ENABLE_NETWORK_OVERRIDE`
```
Comment (zijun726911) [inline docs/configure/environment.md:69]: That can be a better name: `ENABLE_SERVICE_NETWORK_OVERRIDE` ?

**Comment by zijun726911** [comment]
Can you update doc and add instruction says that we need to pass DEFAULT_SERVICE_NETWORK to start the controller when run the `make e2e-test`? would be good if you could run e2e-test for this PR.

Overall LGTM.

File: pkg/deploy/lattice/service_network_manager.go [hunk 2]
```diff
@@ -209,194 +186,30 @@ func (m *defaultServiceNetworkManager) CreateOrUpdate(ctx context.Context, servi
...
-
-		m.log.Debugf("Deleting service network %s", snName)
-		deleteInput := vpclattice.DeleteServiceNetworkInput{
-			ServiceNetworkIdentifier: &serviceNetworkId,
-		}
-		resp, err := vpcLatticeSess.DeleteServiceNetworkWithContext(ctx, &deleteInput)
-		if err != nil {
-			return fmt.Errorf("%w: failed to delete service network %s due to %s", RetryErr, snName, resp)
+		if snva != nil {
+			m.log.Debugf("ServiceNetwork %s also has VPC association %s", serviceNetwork.Spec.Name, snva.Arn)
```
Comment (zijun726911) [inline pkg/deploy/lattice/service_network_manager.go:195]: should be: ServiceNetwork %s  __already__ has VPC association ?

**Comment by solmonk** [comment]
Addressed the comments, thanks

**Review by zijun726911 — APPROVED** [review]

**Pull request merged**